x-pack/plugins/code/server/lsp/request_expander.ts

```diff
@@ -147,7 +150,13 @@ export class RequestExpander implements ILanguageServerHandler {
 
   private async sendInitRequest(workspacePath: string) {
     return await this.proxy.initialize(
-      {},
+      {
```

Comment:
You don't need hard code these here. If you do need to send them as `ClientCapabilities`,  I suggest you do following:

1.define a new `InitializeOptions` as param:
```
interface InitializeOptions {
    clientCapabilities?:ClientCapabilities
    initialOptions?: object
}
```
2. change proxy.ts to use both  `clientCapabilities` and `initialOptions` when send `initialize` request.

3. Then pass your `clientCapabilities` when you create expander for go server.

Reply:
I know these changes a lot, but hard code `ClientCapabilities` may influence all language server s, we can't guarantee all of them can support his.

Reply:
Thanks for your review!
> I know these changes a lot, but hard code `ClientCapabilities` may influence all language server s, we can't guarantee all of them can support his.

That's what I also concerned about. I think it's too much to redefine a new `InitializeOptions` to encapsulate the `ClientCapabilities` and `initialOptions`. I would use `initialOptions` to pass the markup kind just as fuyao said.

Reply:
@spacedragon I have updated the PR according to your suggestion. Please review!